src/main/java/de/zalando/spring/cloud/config/aws/kms/KmsEncryptionConfiguration.java

```diff
@@ -52,20 +54,42 @@ KmsTextEncryptor kmsTextEncryptor() {
             return new KmsTextEncryptor(kms, properties.getKeyId());
         }
     }
+    
+    @Configuration
+    @ConditionalOnProperty(name="aws.kms.endpoint.service-endpoint")
+    @ConditionalOnMissingBean(AWSKMS.class)
+    static class KmsConfigurationEndpoint {
+
+        private final KmsProperties properties;
+
+        @Autowired
+        public KmsConfigurationEndpoint(KmsProperties properties) {
+            this.properties = properties;
+        }
+
+        @Bean
+        public AWSKMS kms() {
+            final AWSKMSClientBuilder builder = AWSKMSClient.builder();
+           	builder.withEndpointConfiguration(new EndpointConfiguration(properties.getEndpoint().getServiceEndpoint(), properties.getEndpoint().getSigningRegion()));
+            return builder.build();
+        }
+
+    }
 
     @Configuration
+    @ConditionalOnProperty(name="aws.kms.region", matchIfMissing=true)
```

Comment:
This doesn't work for me. Spring never picks the `KmsConfigurationEndpoint` class because the region condition is alway true. I would suggest to not use `ConditionalOnProperty` here at all and just handle the business logic the `kms()` bean method, like:

```
if (serviceEndpoint present) {
  builder.withEndpointConfig(serviceEndpoint)
}

Reply:
Fixed.